src/OrchardCore.Modules/OrchardCore.ContentFields/Controllers/ContentPickerAdminController.cs

```diff
@@ -75,7 +86,23 @@ public async Task<IActionResult> SearchContentItems(string part, string field, s
                 PartFieldDefinition = partFieldDefinition
             });
 
-            return new ObjectResult(results.Select(r => new VueMultiselectItemViewModel() { Id = r.ContentItemId, DisplayText = r.DisplayText, HasPublished = r.HasPublished }));
+            var contentItems = await _contentManager
+                .GetAsync(results.Select(r => r.ContentItemId));
+
+            var selectedItems = new List<VueMultiselectItemViewModel>();
+            foreach (var contentItem in contentItems)
+            {
+                selectedItems.Add(new VueMultiselectItemViewModel()
+                {
+                    Id = contentItem.ContentItemId,
+                    DisplayText = contentItem.ToString(),
+                    HasPublished = contentItem.IsPublished(),
+                    IsViewable = await _authorizationService.AuthorizeAsync(_httpContextAccessor.HttpContext!.User,
```

Comment:
`_httpContextAccessor.HttpContext?.User` also here you are checking for `ViewContent` not `EditContent`.